### PR TITLE
opam upgrade caqti

### PR DESCRIPTION
**Pull request by emberian**
The upgraded caqti will do maintenance operations under the hood that should reduce the memory usage we see. Additionally the pool config can now be loaded from `CAQTI_POOL_*` environment variables. nice! hopefully this works, though 😬 

develop merge: https://github.com/MinaProtocol/mina/pull/15012

### CODE REVIEW

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-nightly-me

**Comment by emberian** [comment]
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/1258

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-nightly-me

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
Approved but seems like there're still some update needed for the `rosetta` and `berkeley_migration` app.

**Review by rbonichon — APPROVED** [review]

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-toolchain-me

**Comment by joaosreis** [comment]
!ci-build-me

File: src/lib/crypto/snarky_tests/snarky_tests.ml [hunk 1]
```diff
@@ -185,7 +185,7 @@ module RangeCircuits = struct
     QCheck.Test.make ~count:100
       ~name:"test range gates during witness generation"
       (* TODO: it'd be nicer to generate actual fields directly, since that domain is most likely smaller *)
-      QCheck.(tup3 bool pos_int pos_int)
+      QCheck.(t3 bool pos_int pos_int)
```
Comment (ghost-not-in-the-shell) [inline src/lib/crypto/snarky_tests/snarky_tests.ml:188]: seems like an unwanted test

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
Not taking this for berkeley, it has served its purpose and will be part of the next release.

**Comment by dkijania** [comment]
!ci-build-me

**Pull request closed without merging**